exercises/concept/sorting-room/.meta/exemplar.go

```diff
@@ -0,0 +1,55 @@
+package sorting
+
+import "fmt"
+
+type NumberBox interface {
+	Number() int
+}
+
+type BooleanBox interface {
+	Boolean() bool
+}
+
+// DescribeNumber should return a string describing the number.
+func DescribeNumber(f float64) string {
+	return fmt.Sprintf("This is the number %.1f", f)
+}
+
+// DescribeBoolean should return a string describing the boolean.
+func DescribeBoolean(b bool) string {
+	if b {
+		return "This is true"
+	}
+	return "This is false"
+}
+
+// DescribeNumberBox should return a string describing the NumberBox.
+func DescribeNumberBox(nb NumberBox) string {
+	return fmt.Sprintf("This is a box containing the number %.1f", float64(nb.Number()))
+}
+
+// DescribeBooleanBox should return a string describing the BooleanBox.
+func DescribeBooleanBox(bb BooleanBox) string {
+	if bb.Boolean() {
+		return "This box contains true"
+	}
+	return "This box contains false"
+}
+
+// DescribeAnything should return a string describing whatever it contains.
+func DescribeAnything(i interface{}) string {
```

Comment:
The student does not know about `interface{}` yet since we have no exercise that introduces it. The plan was to combine it with introducing JSON encoding but it might be a while until that exercise will be available. To fix maybe we could add a short bit on this in introduction. E.g. something like "A common use case for type assertion is to find the actual type for values that are specified as empty interface `interface{}` which can potentially hold values of all types."

Reply:
As mentioned in #1645:

> This exercise will have at least `interfaces` as a prerequisite.

I would expect the `interfaces` concept / exercise to introduce the empty interface. Is that not a reasonable assumption? Coincidentally, I also have the issue for teaching `interfaces` in `the-farm` so I could certainly make sure it is mentioned there (even though it will not specifically be used there).

Reply:
In my head `interfaces` (with actual methods) and the empty interface were separate concepts but of course your version also works.